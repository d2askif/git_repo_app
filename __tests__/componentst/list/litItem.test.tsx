import 'react-native';
import renderer from 'react-test-renderer';
import React from 'react';
import ListItem from '../../../src/components/list/listItem';
test('listItem component', () => {
  const props = {
    onStarPressed: jest.fn(),
    avatar_url: 'url',
    name: 'name',
    description: 'description',
    stargazers_count: 1200,
    isStared: false,
    onPress: jest.fn(),
  };
  const snapshot = renderer.create(<ListItem {...props} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
