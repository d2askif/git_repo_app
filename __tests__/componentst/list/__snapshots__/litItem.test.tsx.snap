// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`listItem component 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 16,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "justifyContent": "flex-start",
            "marginRight": 8,
          }
        }
      >
        <View
          activeOpacity={0.7}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 37.5,
              "height": 75,
              "width": 75,
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Object {
                "backgroundColor": "#bdbdbd",
                "borderRadius": 37.5,
                "flex": 1,
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            <Image
              onLoad={[Function]}
              source={
                Object {
                  "uri": "url",
                }
              }
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": null,
                    "width": null,
                  },
                ]
              }
              testID="RNE__Image"
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            />
            <View
              accessibilityElementsHidden={true}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#bdbdbd",
                    "flex": 1,
                    "height": null,
                    "justifyContent": "center",
                    "width": null,
                  }
                }
                testID="RNE__Image__placeholder"
              >
                <Text
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#ffffff",
                      "fontSize": 37.5,
                      "textAlign": "center",
                    }
                  }
                >
                  n
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "height": null,
                  "width": null,
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "paddingLeft": 8,
            "width": 434.99999999999994,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "grey",
                "fontSize": 15,
                "fontWeight": "bold",
                "paddingBottom": 8,
              }
            }
          >
            name
          </Text>
        </View>
        <View>
          <Text
            numberOfLines={3}
            style={
              Object {
                "color": "#ACACC1",
                "fontSize": 14,
                "letterSpacing": 1,
                "textAlign": "left",
              }
            }
          >
            description
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingTop": 16,
        }
      }
    >
      <View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "lightgrey",
                  "fontSize": 20,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="iconIcon"
          >
            ï€…
          </Text>
        </View>
      </View>
      <Text>
         
        1.2K
      </Text>
    </View>
  </View>
</View>
`;
